app/src/main/java/au/id/neasbey/biblecast/API/BibleAPIResponseHandler.java

```diff
@@ -0,0 +1,135 @@
+package au.id.neasbey.biblecast.API;
+
+import android.text.Spanned;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.util.LinkedList;
+import java.util.List;
+
+import au.id.neasbey.biblecast.R;
+import au.id.neasbey.biblecast.util.UIUtils;
+
+/**
+ * Created by craigneasbey on 11/08/15.
+ *
+ * Handles the the response from the Bible API
+ */
+public abstract class BibleAPIResponseHandler {
```

Comment:
I find this a bit confusing. Perhaps this should just be a HttpResponse ... something generic (pre-built ones might exist in a different http library) that simply gets all of the http response data (status codes, response headers and message body).

The Parser part should be moved into the parser class (and so this object could be de-coupled from the parser)